Implementation of pointcut builder with closures

lisachenko opened the pull request:
For annotation haters, AOP is awful slow non-intuitive tool. There are a lot of requests to support native closures for definition of advices.

This PR aims to give a tool for defining advices with native PHP syntax.

Example:

``` php
$pointcutBuilder = new PointcutBuilder($container);
$pointcutBuilder->before('execution(public **->*(*))', function (MethodInvocation $method) {
    echo $method, PHP_EOL;
});

$pointcutBuilder->after('access(* Demo\**->*)', function (ClassFieldAccess $property) {
    echo $property, PHP_EOL;
});
```

kamilwylegala commented:
Is it going to be merged?

lisachenko commented:
Yes, sure, but no one asks me about merging it :) You are first! Is this feature important for you?

kamilwylegala commented:
Yes, I'd be glad :) I'm trying to adapt go-aop-php in my project but I'm stuck in aspect definition (Cacheable feature). So I would like to try approach from this merge request.

lisachenko committed:
Implementation of pointcut builder with closures.

Implemented general pointcuts: before, after, around, afterThrowing, declareError
